test/functional/pdo_sqlsrv/pdo_018_next_result_set.phpt

```diff
@@ -1,56 +1,85 @@
 --TEST--
 Moves the cursor to the next result set
 --SKIPIF--
-<?php require('skipif.inc'); ?>
+<?php require('skipif_mid-refactor.inc'); ?>
 --FILE--
 <?php
-require_once("MsSetup.inc");
-
-/* Connect */
-$conn = new PDO("sqlsrv:server=$server; database=$databaseName", $uid, $pwd);
-
-// Create table
-$tableName = '#testResultSet';
-$sql = "CREATE TABLE $tableName (c1 INT, c2 VARCHAR(40))";
-$stmt = $conn->exec($sql);
-
-// Insert data using bind parameters
-$sql = "INSERT INTO $tableName VALUES (?,?)";
-for($t=200; $t<220; $t++) {
-	$stmt = $conn->prepare($sql);
-	$stmt->bindParam(1, $t);
-	$ts = sha1($t);
-	$stmt->bindParam(2, $ts);
-	$stmt->execute();
-}
+require_once("MsCommon_mid-refactor.inc");
+
+try {
+    /* Connect */
+    $conn = connect();
 
-// Fetch, get data and move the cursor to the next result set
-$sql = "SELECT * from $tableName WHERE c1 = '204' OR c1 = '210'; 
-		SELECT Top 3 * FROM $tableName ORDER BY c1 DESC";
-$stmt = $conn->query($sql);
-$data1 = $stmt->fetchAll(PDO::FETCH_ASSOC);
-$stmt->nextRowset();
-$data2 = $stmt->fetchAll(PDO::FETCH_NUM);
+    // Create table
+    $tableName = 'testResultSet';
+    createTable($conn, $tableName, array("c1" => "int", "c2" => "varchar(40)"));
 
-// Array: FETCH_ASSOC
-foreach ($data1 as $a)
-echo $a['c1'] . "|" . $a['c2'] . "\n";
+    // Insert data using bind parameters
+    $sql = "INSERT INTO $tableName VALUES (?,?)";
+    $inputs = array();
+    for ($t=200; $t<220; $t++) {
+        $stmt = $conn->prepare($sql);
+        $stmt->bindParam(1, $t);
+        $ts = sha1($t);
+        $stmt->bindParam(2, $ts);
+        $stmt->execute();
+        array_push($inputs, array($t, $ts));
+    }
 
-// Array: FETCH_NUM
-foreach ($data2 as $a)
-echo $a[0] . "|" . $a[1] . "\n";
+    // Fetch, get data and move the cursor to the next result set
+    if (!isColEncrypted()) {
+        $sql = "SELECT * from $tableName WHERE c1 = '204' OR c1 = '210'; 
+                SELECT Top 3 * FROM $tableName ORDER BY c1 DESC";
+        $stmt = $conn->query($sql);
+        $expected = array(array(219, sha1(219)), array(218, sha1(218)), array(217, sha1(217)));
+    } else {
+        // ORDER BY does not work for encrypted columns
+        $sql = "SELECT * FROM $tableName WHERE c1 = ? OR c1 = ?;
```

Comment:
There's not explicit documentation that says ORDER BY doesn't work on encrypted column, but in https://docs.microsoft.com/en-us/sql/relational-databases/security/encryption/always-encrypted-database-engine, the Feature Details section states that operators such as greater/less than does not work for encrypted columns, and ORDER BY is based on that. I also tried `SELECT Top 3 * FROM $tableName ORDER BY c1 DESC` on an encrypted column in SSMS, the order they return is completely different from expected.

Where clause to restrict c1 < 220 and >= 217 doesn't work either since greater/less than operators are not supported for encrypted columns.

Reply:
It's alright then. Please write a long comment and provide the reference links.

Reply:
ok